pkg/blob/fetch.go

```diff
@@ -105,3 +137,36 @@ func classifyFile(reader io.ReadSeeker) (string, error) {
 
 	return http.DetectContentType(buf), nil
 }
+
+func sha256sum(reader io.ReadSeeker) (string, error) {
+	hash := sha256.New()
+	_, err := io.Copy(hash, reader)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = reader.Seek(0, 0)
```

Comment:
Nice! :) I would have assumed the reader was zero'ed and then spent hours debugging this when the inevitable bug report landed.